test/tar_interpreter_test.go

```diff
@@ -0,0 +1,156 @@
+package test
+
+import (
+	"archive/tar"
+	"bytes"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wal-g/wal-g/internal"
+)
+
+type testInterpretOptions struct {
+	DBDataDirectory string
+
+	Name     string
+	Typeflag byte
+
+	Create func(string) error
+	Delete func(string) error
+}
+
+func testInterpret(t *testing.T, options testInterpretOptions) (os.FileInfo, os.FileInfo) {
+	tarInterpreter := &internal.FileTarInterpreter{
+		DBDataDirectory: options.DBDataDirectory,
+	}
+
+	err := options.Create(options.Name)
+
+	assert.NoError(t, err)
+
+	defer options.Delete(options.Name)
```

Comment:
It looks like an unhandled error. Try to replace it with something similar to this https://pastebin.com/7xAFJtpX